include/particle.tcc

```diff
@@ -242,6 +242,137 @@ void mpm::Particle<Tdim>::initialise() {
   this->properties_["displacements"] = [&]() { return displacement(); };
 }
 
+//! Clone material state variables from neighbour particle
+template <unsigned Tdim>
+bool mpm::Particle<Tdim>::clone_neighbour_material_state_vars(
+    const mpm::dense_map& state_vars,
+    const std::shared_ptr<mpm::Material<Tdim>>& material) {
+  bool status = true;
+  if (material != nullptr) {
+    if (this->material_id_ == material->id() ||
+        this->material_id_ == std::numeric_limits<unsigned>::max()) {
+      material_ = material;
+      // Clone state variables
+      auto mat_state_vars = material_->initialise_state_variables();
+      if (mat_state_vars.size() == state_vars.size()) {
+        this->state_variables_ = state_vars;
+      } else {
+        status = false;
+        throw std::runtime_error(
+            "# of state variables do not match! Cannot clone material history");
+      }
+    } else {
+      status = false;
+      throw std::runtime_error(
+          "Particle material is invalid! Cannot assign a new material");
+    }
+  } else {
+    status = false;
+    throw std::runtime_error(
+        "Material is invalid! cannot clone material history");
+  }
+  return status;
+}
+
+// Interpolate properties from neighbour particles
+// TODO: how to pass the interpolator (can be MLS or radial basis function)
+// TODO: check other properties
+// TODO: check exception handling
+template <unsigned Tdim>
+bool mpm::Particle<Tdim>::interpolate_neighbour_properties(
```

Comment:
Shouldn't we be doing this at a mesh level?

Reply:
I would agree to do it from mesh, as particles dont have pointer to its neighbour

Reply:
@kks32 Shall we discuss this part more carefully as I have few things to clarify?

Reply:
Moved to mesh.